src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -90,6 +90,53 @@ class Wavefunction
 
     void flush() const
     {
+        //logic to reorder
+        Fusion fg = fused_.getFusedGates();
+        auto items = fg.getItems();
+        auto sets = fg.getSet();
+        auto ctrlSets = fg.getCtrlSet();
+        //getting all qubits to move to lower end of the wfn
+        if (items.size() > 0) {
+            std::vector<unsigned> indices;
+            std::set<unsigned> indicesSet;
+            for (int i = 0; i < items.size(); i++) {
+                auto tempIndices = items[i].get_indices();
+                for (unsigned j = 0; j < tempIndices.size(); j++) {
+                    if (indicesSet.count(tempIndices[j]) == 0) {
+                        indices.push_back(tempIndices[j]);
```

Comment:
it would be more efficient to dump indicesSet into a vector at once after the union set is populated (push_back reallocates the vector as it has to expand it): `indices.assign(indicesSet.begin(), indicesSet.end());`

Reply:
The reason this approach is being used is because we want to preserve the original order of the qubits in the target and control sets. Dumping the set into the vector will result in them being scrambled into an ascending order. Even if we switch to an unordered set it will not guarantee the preservation of the original order

Reply:
Please document this requirement with a comment because it's not obvious from the code, and someone might later attempt refactors that would violate it. To avoid the mltiple re-allocations of the indices vector, consider reserving enough space upfront.